### ISSUE TITLE
Lint step failing 

### ISSUE THREAD

**Issue by Nemixdre** [kimdv/bitrise-step-swiftlint#15]
Hi.

We have a problem with the linter step for 2-3w roughly. 
On our repo we have a branch on which this step wasn't problematic on the 2nd of November: 
We haven't touched this branch (no new commit) since then as I've been in holidays. Now that I'm back, the same flow doesn't work anymore and fails on the linter step SwiftLint.

The log is now showing violations and stuff it didn't use to bring up:
```
Done linting! Found 451 violations, 10 serious in 862 files.
|                                                                              |
+---+---------------------------------------------------------------+----------+
| x | swiftlint@0.8 (exit code: 2)                                  | 23.66 sec|
+---+---------------------------------------------------------------+----------+
| Issue tracker: https://github.com/kimdv/bitrise-step-swiftlint/issues        |
| Source: https://github.com/kimdv/bitrise-step-swiftlint                      |
+---+---------------------------------------------------------------+----------+
```

Before:
```
Done linting! Found 2 violations, 0 serious in 862 files.
Saved swiftlint output in SWIFTLINT_REPORT
Saved swiftlint output in file at path SWIFTLINT_REPORT_PATH
|                                                                              |
+---+---------------------------------------------------------------+----------+
| ✓ | swiftlint@0.8                                                 | 24.01 sec|
+---+---------------------------------------------------------------+----------+
```

To me it looks like the linter isn't taking our swiftlint.yml file as a reference for the linting anymore but the default one.

Bitrise support recommended me to contact you as they couldn't find anything in their stack changes that could have had this impact.

Can you help please?

Thanks.

**Comment by danielrbrowne** [comment]
I can confirm I've seen similar behaviour - some time on or after 12th November, I believe there has been a server-side change to how Bitrise handles nested `.swiftlint.yml` config files. **With zero code changes, I've seen our SwiftLint step begin failing due to reporting violations that should not be reported if our nested config YAML files were being respected.**

In terms of diagnosing this issue, this is what I did:
- First, I tested locally by running `swiftlint` on my local working copy using the latest release of SwiftLint (0.41.0). This reported the same number of violations as I expected, and correctly supressing violations based on our nested YAML config files.
- I pushed a branch adding the `Test` directory in my repo to my root `.swiftlint.yml` file's `excluded:` list. This allowed the Bitrise SwiftLint step to pass once again. (Since the whole directory was excluded, which otherwise contains a nested config file disabling only specific SwiftLint rules for that directory).

The reasons I suggest that this is a server-side breaking change:
- No new version of the Bitrise SwiftLint CI step has been released since the latest 0.8.0 release in July.
- I tried switching to an older version of the Bitrise SwiftLint CI step, and saw the same erroneous behaviour regarding nested YAML config files being ignored.

I've opened a support ticket two days ago and I'm still waiting for a reply.

**Comment by kimdv** [comment]
Hi @Nemixdre ! 

Thanks for reaching out. 

I agree with @danielrbrowne that it seems strange it started breaking out of nothing. 
Let's wait and see what the response @danielrbrowne get from the Bitrise team. 🤞

**Comment by danielrbrowne** [comment]
@kimdv / @Nemixdre 

It seems this was a mislabelled breaking change in 0.41.0 of SwiftLint (see this comment, and the issue as a whole: https://github.com/realm/SwiftLint/issues/3417#issuecomment-723728003).

TL;DR Now, if a `--config` path is explicitly provided, searching for nested config files is disabled.

Given this, I simply deleted the default config file path of `$BITRISE_SOURCE_DIR/.swiftlint.yml` in my Bitrise SwiftLint step to resolve my issues.

@kimdv I'd therefore recommend a note is added in the Bitrise step UI for the `lint_config_file` parameter RE: if the dev is using nested configs that they want to be respected.

I hope this info helps!

**Comment by Nemixdre** [comment]
Sorry for the late reply, we had other tech problems to attend.

I tried your solution @danielrbrowne and it works perfectly! Thank you!
